server/controllers/tripController.js

```diff
@@ -60,6 +60,38 @@ class TripController {
     util.setSuccess(200, message, data)
     return util.send(res);
   }
+
+  static async updateOpenTripDetails (req, res){
+    
+    const newTripData = {
+        origin: req.body.origin || req.oldTripDetails.origin,
+        destination: req.body.destination || req.oldTripDetails.destination,
+        dateOfTravel: req.body.date || req.oldTripDetails.dateOfTravel,
+        dateOfReturn: req.body.returnDate || req.oldTripDetails.dateOfReturn,
+        travelReason: req.body.travelReason || req.oldTripDetails.travelReason,
+        accommodation_id: req.body.accommodationID || req.oldTripDetails.accommodation_id,
+        type: req.tripType,
+    }
+    
+
+    const tripDetailsUpdater = await tripService.updateTrip(newTripData, { id: req.params.trip_id })
```

Comment:
@irfiacre what if the updateTrip function fails? It's always to catch errors that might occur when interacting/accessing with an external service (in this instance the database). Applicable elsewhere it is necessary.

Reply:
I think you can only pass around with what user need to update only and perform verification on backend instead of always transmitting not needed data except maybe for special cases

Reply:
@Francois-MUGOROZI I do not understand your first comment if you could make it clear for me it could be better.